### DISCUSSION TITLE
Is there any function to add an item before/after a specific item in a collection

### DISCUSSION THREAD

**Discussion by mdariftiens** [laravel/framework#51749] [Q&A]
I need to add an item after an item in a collection. Now I'm doing like below. is there any function to do this? Or can I make a PR for this functionality? 

```
$collection = collect(['first', 'second', 'third', 'fourth']);

$position = $collection->search('second');

$collection->splice($position + 1, 0, [$newItem]);
```

**Comment by ennorehling** [comment]
I have a similar problem right now, and while looking for a solution came across search and splice, but: search returns a key, and splice expects an index. This leads to unexpected results:
```
$c = collect([1=>1, 3 =>3]);
$key = c->search(fn($i) $i > 1); // returns 3
$c->splice($key, 0, [4=>4, 5=>5]);
dump($c->all()); // prints [1, 3, 4, 5]
```

**Comment by maulanaa121** [comment]
You can use `splice()` with the position returned by `search()`.

For example:

```php
$position = $collection->search('second');

$collection->splice($position + 1, 0, [$newItem]);

**Comment by maulanaa121** [comment]
I don't think you should manually verify the Passport access token using `hash_hmac()`.

Laravel Passport already handles access-token validation through its authentication guard. You can protect your API route with Passport:

```php
Route::get('/user', function (Request $request) {
    return $request->user();
})->middleware('auth:api');
```

Then send the token as a Bearer token:

```http
Authorization: Bearer YOUR_ACCESS_TOKEN
```

Passport will handle the token parsing and signature/expiration/revocation checks for you.

If you really need to inspect or verify the JWT manually, you should use a JWT library and verify it using Passport's configured public key and the algorithm specified by the token, rather than recreating the signature with `hash_hmac('SHA256', ...)`.

So the main issue with the current implementation is that it is trying to reproduce Passport's token verification logic manually.

**Reply by henzeb** [reply]
an you delete this post? it does not belong here.

**Comment by nakgamers** [comment]
Great catch from @ennorehling — this is exactly why you should not pass the result of `search()` straight into `splice()`. `search()` returns the **key** of the matched item, while `splice()` expects a **numeric index** (0-based offset). They are only interchangeable when the collection is a plain list with sequential integer keys.

The robust way is to convert the key to its positional index first:

```php
$c = collect([1 => 1, 3 => 3]);

$key   = $c->search(fn ($i) => $i > 1); // 3  -> the KEY
$index = $c->keys()->search($key);      // 1  -> its position (0-based)

$c->splice($index + 1, 0, [4 => 4]);

dump($c->all()); // [1 => 1, 3 => 3, 4 => 4]  -> inserted AFTER item with key 3
```

If you do not care about preserving the original keys, you can also reindex first with `values()` so that key === index, and then splice as in the original snippet:

```php
$c = collect([1 => 1, 3 => 3])->values();   // [0 => 1, 1 => 3]
$pos = $c->search(fn ($i) => $i > 1);        // 1 (now it IS the index)
$c->splice($pos + 1, 0, [4 => 4]);
dump($c->all()); // [0 => 1, 1 => 3, 2 => 4]
```

There is still no built-in `insertAfter()`/`insertBefore()` method, so `keys()->search()` + `splice()` (or a `values()` reindex) remains the idiomatic workaround. Worth mentioning that `splice()` **reindexes numeric keys** after the operation, so if the collection is associative you want the `keys()->search()` variant to keep your string keys intact.
